modules/apps/portlet-configuration/portlet-configuration-web/src/main/resources/META-INF/resources/edit_permissions.jsp

```diff
@@ -323,4 +329,66 @@ PortletConfigurationPermissionPropagation portletConfigurationPermissionPropagat
 			}
 		});
 	}
+
+	function openPopUp(href) {
+		Liferay.Util.openModal({
+			bodyHTML:
+				'<liferay-ui:message key="changing-tab-without-save-helper" />',
+			buttons: [
+				{
+					autoFocus: true,
+					displayType: 'secondary',
+					label: '<liferay-ui:message key="cancel" />',
+					type: 'cancel',
+				},
+				{
+					displayType: 'secondary',
+					label: '<liferay-ui:message key="discard" />',
+					onClick: () => {
+						window.location.href = href;
+					},
+				},
+				{
+					displayType: 'warning',
+					label: '<liferay-ui:message key="save-and-continue" />',
+					onClick: () => {
+						<portlet:namespace />saveButton.dispatchEvent(
+							new Event('click')
+						);
+						window.location.href = href;
+					},
+				},
+			],
+			status: 'warning',
+			title: '<liferay-ui:message key="discard-changes" />' + '?',
+		});
+	}
+
+	var propagationNavigationBar = document.getElementById(
+		'<portlet:namespace />propagationNavigationBar'
+	);
+
+	if (propagationNavigationBar) {
+		propagationNavigationBar.addEventListener('click', (event) => {
+			if (event.target.tagName === 'SPAN' || event.target.tagName === 'A') {
+				event.preventDefault();
+				const target =
+					event.target.tagName === 'A'
+						? event.target
+						: event.target.parentElement;
+
+				var count = document.getElementById('<portlet:namespace />count');
+
+				if (
+					!target.classList.contains('active') &&
+					Number(count.value) !== 0
+				) {
+					openPopUp(target.href);
+				}
+				else {
+					window.location.href = target.href;
+				}
```

Comment:
I think we can move `event.preventDefault()` to `if` block instead. There is no need to add an `else` block.